tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/event/RabbitMQAndRedisEventBusModule.java

```diff
@@ -81,10 +83,15 @@ sender, receiverProvider, eventSerializer, retryBackoffConfiguration, new Routin
     }
 
     @ProvidesIntoSet
-    InitializationOperation workQueue(RabbitMQAndRedisEventBus instance) {
+    InitializationOperation workQueue(RabbitMQAndRedisEventBus instance, CleanRedisEventBusService cleanRedisEventBusService) {
         return InitilizationOperationBuilder
             .forClass(RabbitMQAndRedisEventBus.class)
-            .init(instance::start);
+            .init(() -> {
+                instance.start();
+                cleanRedisEventBusService.cleanUp()
+                    .subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
```

Comment:
Why blocvking call wrapper?

Reply:
To avoid any cleanup code could run on the TMail startup thread and make the startup time slower.